root/etc/e-smith/events/actions/nethserver-mattermost-conf

```diff
@@ -20,21 +20,59 @@
 # along with NethServer.  If not, see COPYING.
 #
 
+su - postgres -c "scl enable rh-postgresql12 -- psql --port=55434 -lqt | cut -d \| -f 1 | grep -q -w mattermost"
+if [[ $? -eq 0 ]]; then # mattermost database exists, exiting
+    exit 0
+fi
 su - postgres -c "scl enable rh-postgresql94 -- psql --port=55432 -lqt | cut -d \| -f 1 | grep -q -w mattermost"
-if [ $? -eq 1 ]; then # database do not exists
-
+if [[ $? -eq 0 ]]; then
+    echo "[NOTICE] Migrating data from postgresql 9.4 to postgresql 12"
+    databases=$(su - postgres -c "scl enable rh-postgresql94 -- psql -q -A -t --port=55432 -c $'SELECT datname FROM pg_database WHERE datistemplate = false AND datname <> \'postgres\';'")
+    if [[ $? -ne 0 ]]; then
+        echo "[ERROR] An error occured while retrieving list of databases"
+        exit 1
+    fi
+    systemctl stop rh-postgresql94-postgresql
+    systemctl stop rh-postgresql12-postgresql
+    su - postgres -c "export LD_LIBRARY_PATH=/opt/rh/rh-postgresql94/root/usr/lib64:/opt/rh/rh-postgresql12/root/usr/lib64:$LD_LIBRARY_PATH && scl enable rh-postgresql12 -- pg_upgrade -b /opt/rh/rh-postgresql94/root/usr/bin -B /opt/rh/rh-postgresql12/root/usr/bin -d /var/opt/rh/rh-postgresql94/lib/pgsql/data -D /var/opt/rh/rh-postgresql12/lib/pgsql/data"
+    if [[ $? -ne 0 ]]; then
+        echo "[ERROR] An error occured during postgresql migration"
```

Comment:
Let's try to recover the initial DB state.

```suggestion
        systemctl start rh-postgresql94-postgresql rh-postgresql12-postgresql
        echo "[ERROR] An error occured during postgresql migration"
```

We are assuming the initial services state is "running". We must add an initial check: if they aren't both running this action shouldn't run IMO.

Reply:
I agree, this could recover from a failed state